.github/workflows/test.yml

```diff
@@ -18,15 +18,27 @@ jobs:
           - "3.9"
           - "3.10"
           - "3.11"
+        test_type:
+          - "core"
+          - "extended"
+    name: Python ${{ matrix.python-version }} ${{ matrix.test_type }}
     steps:
       - uses: actions/checkout@v3
       - name: Set up Python ${{ matrix.python-version }}
         uses: "./.github/actions/poetry_setup"
         with:
           python-version: ${{ matrix.python-version }}
           poetry-version: "1.4.2"
-          cache-key: "main"
-          install-command: "poetry install"
-      - name: Run unit tests
+          cache-key: ${{ matrix.test_type }}
+          install-command: |
+              if [ "${{ matrix.test_type }}" == "core" ]; then
+                echo "Running core tests, installing dependencies with poetry..."
+                poetry install
+              else
+                echo "Running extended tests, installing dependencies with poetry..."
+                poetry install -E extended_testing
+              fi
+      - name: Run $${{matrix.test_type}} tests
         run: |
-          make test
+          make tests
```

Comment:
why the rename, and why do we have both?

Reply:
as in why rename here, and why do we have both in makefile

Reply:
Oh rename is totally accidental, but it does the same thing. I'll follow up with a minor typo fix in a second after landing this.